eduNEXT/seb-openedx
# Adding support from Koa to Olive versions in legacy mode

This PR has the intention to make some changes that will allow users of the seb browser plugin to install it in versions later than hawthorn, so far the correct operation from Koa to Olive version in legacy mode has been tested, to achieve this goal the eduNEXT team made the following modifications:

- We removed the previous CI that was in CircleCI and added a new set using Github Actions, this test set was implemented to test unit tests in both Python 3.6 and 3.8.
-  Python 2.7 support was removed and only python 3 support was left.
- New backends were added for nutmeg, these backends were tested and work correctly for versions from Koa to Olive.
- Updated the version of the unit test dependencies to match those used by the platform in nutmeg.

### How to run unit tests locally

1. To run the unit tests, a Python virtual environment must be created.
`mkvirtualenv venv`
2. Activate your virtual environment
`source venv/bin/activate`
3. Install unit test requirements.
`pip install -r requirements.txt`
4. Run the tests.
`coverage run --source="." -m pytest`

### How to test the plugin functionalities manually

1. Create a yaml tutor plugin and activate it, to know further details about ysml tutor plugins, read the following article [YAML file](https://docs.tutor.overhang.io/plugins/v0/gettingstarted.html#yaml-file).
``` yaml
name: seb-backend-plugin
version: 0.1.0
patches:
  openedx-common-settings: |
    MIDDLEWARE.append("seb_openedx.middleware.SecureExamBrowserMiddleware")

  common-env-features: |
    ENABLE_OTHER_COURSE_SETTINGS: true
```
2. Install the seb-openedx plugin as an extra requirement for more information on how to install an extra requirement in tutor you should follow the article below [Installing extra xblocks and requirements](https://docs.tutor.overhang.io/configuration.html#installing-extra-xblocks-and-requirements).
3. Configure the plugin, in the following link there is useful information for its correct configuration [seb-openedx documentation](https://seb-openedx.readthedocs.io/).
